src/tree/traits.rs

```diff
@@ -0,0 +1,127 @@
+//! The abstractions that make up the core of Taffy's low-level API
```

Comment:
A high level explanation of these traits and how they interrelate would be really useful here.

Reply:
That is in fact why I created the module! I just ran out of steam before I got to actually writing the documentation.